### PR TITLE
Im 187 frontend routing

### PR DESCRIPTION
**Pull request by NamespaceValentine**

### CODE REVIEW

File: public/javascripts/DisplayInventory.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+                if (e.which == 13) {
+                    self.searchItem($("#item").val());
+                }
+            });
+
+            $("#search_item_button").click(function() {
+                self.searchItem($("#item").val());
+            });
+
+            var query = window.location.search.substring(1).split('&', 1);
```
Comment (NamespaceValentine) [inline public/javascripts/DisplayInventory.js:89]: This line to 110 are changes I made to this file.

**Comment by mrstarware** [comment]
Github says there's conflicts, and you may need to refactor $.getJson to a navigation.hit 
Otherwise I think it looks good!

**Comment by NamespaceValentine** [comment]
Until hit is refactored for the use case I need, I require using jquery.get for this. The reason being, my code currently requires the use of .fail, which is not given to me through navigation.hit

**Comment by NamespaceValentine** [comment]
Branch now contains the qr changes

**Pull request merged**